File: vacgom-core/src/main/java/kr/co/vacgom/error/CommonErrorResponse.java
package kr.co.vacgom.error;

import lombok.Getter;

import java.time.LocalDateTime;

@Getter
public class CommonErrorResponse {

    private final String timeStamp;

    private final String errorCode;

    private final String errorMessage;

    private CommonErrorResponse(
            String errorCode,
            String errorMessage
    ) {
        this.timeStamp = LocalDateTime.now().toString();
        this.errorCode = errorCode;
        this.errorMessage = errorMessage;
    }

    public static CommonErrorResponse from(ErrorEntity errorEntity) {
        return new CommonErrorResponse(errorEntity.getCode(), errorEntity.getMessage());
    }

    public static CommonErrorResponse of(ErrorEntity errorEntity, String errorMessage) {
        return new CommonErrorResponse(errorEntity.getCode(), errorMessage);
    }
}

File: vacgom-common/src/main/kotlin/kr/co/vacgom/common/error/dto/CommonErrorResponse.kt
package kr.co.vacgom.common.error.dto

import kr.co.vacgom.common.error.entity.ErrorEntity
import java.time.LocalDateTime

data class CommonErrorResponse(
    val timeStamp: String = LocalDateTime.now().toString(),
    val errorCode: String,
    val errorMessage: String?
) {
    constructor(errorEntity: ErrorEntity) : this(
        errorCode = errorEntity.code,
        errorMessage = errorEntity.message
    )

    constructor(
        errorEntity: ErrorEntity,
        errorMessage: String
    ) : this(
        errorCode = errorEntity.code,
        errorMessage = errorMessage
    )
}
